AuthenticatorShared/Core/Platform/Services/API/ConfigAPIService.swift

```diff
@@ -0,0 +1,15 @@
+import Networking
+
+/// A protocol for an API service used to make config requests.
+///
+protocol ConfigAPIService {
+    /// Performs an API request to get the configuration from the backend.
+    ///
+    func getConfig() async throws -> ConfigResponseModel
+}
+
+class DefaultConfigAPIService: ConfigAPIService {
+    func getConfig() async throws -> ConfigResponseModel {
+        throw BitwardenError.dataError("Remote configuration not yet implemented")
+    }
+}
```

Comment:
🤔 Unless the networking layer is getting added very soon I would avoid adding this `ConfigApiService` protocol and default implementation that has no implementation. This could confuse people reading the `ConfigService` tending to think there's some `ConfigApiService` where in fact there's no actual implementation. IMO it would be cleaner to add this api service when there's actually an API call happening.

Reply:
🤦🏼‍♀️ I originally had the `ConfigService` using this, then took that out, and didn't think to swing back through on the `APIService`. Agreed, this should be pretty easy to take out, and we can bring it in correctly if we need it later.